### PR TITLE
WIP: First attempt at Python 3.7 support

### LINKED ISSUE #63: Unable to install pmdarima with python 3.7
pip install pmdarima  --user
Collecting pmdarima
  Using cached https://files.pythonhosted.org/packages/dd/43/d204f675a1258e60fae438205c7019ad4909622982330925a668c7f1f9ae/pmdarima-1.0.0.tar.gz
Requirement already satisfied: Cython>=0.23 in /home/rejuna15/.local/lib/python3.7/site-packages (from pmdarima) (0.27.3)
Requirement already satisfied: numpy>=1.10 in /home/rejuna15/.local/lib/python3.7/site-packages (from pmdarima) (1.15.4)
Requirement already satisfied: pandas>=0.19 in /home/rejuna15/.local/lib/python3.7/site-packages (from pmdarima) (0.23.4)
Requirement already satisfied: scikit-learn>=0.17 in /home/rejuna15/.local/lib/python3.7/site-packages (from pmdarima) (0.20.1)
Requirement already satisfied: scipy>=0.9 in /home/rejuna15/.local/lib/python3.7/site-packages (from pmdarima) (1.1.0)
Requirement already satisfied: statsmodels>=0.9.0 in /home/rejuna15/.local/lib/python3.7/site-packages (from pmdarima) (0.9.0)
Requirement already satisfied: python-dateutil>=2.5.0 in /home/rejuna15/.local/lib/python3.7/site-packages (from pandas>=0.19->pmdarima) (2.7.5)
Requirement already satisfied: pytz>=2011k in /usr/lib/python3/dist-packages (from pandas>=0.19->pmdarima) (2018.3)
Requirement already satisfied: patsy in /home/rejuna15/.local/lib/python3.7/site-packages (from statsmodels>=0.9.0->pmdarima) (0.5.1)
Requirement already satisfied: six>=1.5 in /usr/lib/python3/dist-packages (from python-dateutil>=2.5.0->pandas>=0.19->pmdarima) (1.11.0)
Building wheels for collected packages: pmdarima
  Running setup.py bdist_wheel for pmdarima ... error
  Complete output from command /usr/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-ioa_i29w/pmdarima/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-knh_gr9k --python-tag cp37:
  Partial import of pmdarima during the build process.
  Requirements: ['numpy>=1.10', 'Cython>=0.23', 'scipy>=0.9', 'scikit-learn>=0.17', 'pandas>=0.19', 'statsmodels>=0.9.0']
  Adding extra setuptools args
  /usr/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'long_description_content_type'
    warnings.warn(msg)
  Traceback (most recent call last):
    File "<string>", line 1, in <module>
    File "/tmp/pip-install-ioa_i29w/pmdarima/setup.py", line 248, in <module>
      do_setup()
    File "/tmp/pip-install-ioa_i29w/pmdarima/setup.py", line 244, in do_setup
      setup(**metadata)
    File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/core.py", line 135, in setup
      config = configuration()
    File "/tmp/pip-install-ioa_i29w/pmdarima/setup.py", line 163, in configuration
      config.add_subpackage(DISTNAME)
    File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 1037, in add_subpackage
      caller_level = 2)
    File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 1006, in get_subpackage
      caller_level = caller_level + 1)
    File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 919, in _get_configuration_from_setup_py
      ('.py', 'U', 1))
    File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/compat/py3k.py", line 122, in npy_load_module
      return importlib.machinery.SourceFileLoader(name, fn).load_module()
    File "<frozen importlib._bootstrap_external>", line 407, in _check_name_wrapper
    File "<frozen importlib._bootstrap_external>", line 907, in load_module
    File "<frozen importlib._bootstrap_external>", line 732, in load_module
    File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
    File "<frozen importlib._bootstrap>", line 696, in _load
    File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
    File "<frozen importlib._bootstrap_external>", line 728, in exec_module
    File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
    File "pmdarima/setup.py", line 11, in <module>
      from pmdarima._build_utils import maybe_cythonize_extensions
  ModuleNotFoundError: No module named 'pmdarima._build_utils'
  
  ----------------------------------------
  Failed building wheel for pmdarima
  Running setup.py clean for pmdarima
Failed to build pmdarima
Installing collected packages: pmdarima
  Running setup.py install for pmdarima ... error
    Complete output from command /usr/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-ioa_i29w/pmdarima/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-37l5bnlx/install-record.txt --single-version-externally-managed --compile --user --prefix=:
    Partial import of pmdarima during the build process.
    Requirements: ['numpy>=1.10', 'Cython>=0.23', 'scipy>=0.9', 'scikit-learn>=0.17', 'pandas>=0.19', 'statsmodels>=0.9.0']
    Adding extra setuptools args
    /usr/lib/python3.7/distutils/dist.py:274: UserWarning: Unknown distribution option: 'long_description_content_type'
      warnings.warn(msg)
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-ioa_i29w/pmdarima/setup.py", line 248, in <module>
        do_setup()
      File "/tmp/pip-install-ioa_i29w/pmdarima/setup.py", line 244, in do_setup
        setup(**metadata)
      File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/core.py", line 135, in setup
        config = configuration()
      File "/tmp/pip-install-ioa_i29w/pmdarima/setup.py", line 163, in configuration
        config.add_subpackage(DISTNAME)
      File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 1037, in add_subpackage
        caller_level = 2)
      File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 1006, in get_subpackage
        caller_level = caller_level + 1)
      File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/distutils/misc_util.py", line 919, in _get_configuration_from_setup_py
        ('.py', 'U', 1))
      File "/home/rejuna15/.local/lib/python3.7/site-packages/numpy/compat/py3k.py", line 122, in npy_load_module
        return importlib.machinery.SourceFileLoader(name, fn).load_module()
      File "<frozen importlib._bootstrap_external>", line 407, in _check_name_wrapper
      File "<frozen importlib._bootstrap_external>", line 907, in load_module
      File "<frozen importlib._bootstrap_external>", line 732, in load_module
      File "<frozen importlib._bootstrap>", line 265, in _load_module_shim
      File "<frozen importlib._bootstrap>", line 696, in _load
      File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
      File "<frozen importlib._bootstrap_external>", line 728, in exec_module
      File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
      File "pmdarima/setup.py", line 11, in <module>
        from pmdarima._build_utils import maybe_cythonize_extensions
    ModuleNotFoundError: No module named 'pmdarima._build_utils'
    
    ----------------------------------------
Command "/usr/bin/python3.7 -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-ioa_i29w/pmdarima/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-record-37l5bnlx/install-record.txt --single-version-externally-managed --compile --user --prefix=" failed with error code 1 in /tmp/pip-install-ioa_i29w/pmdarima/

#### Description
Works with python 3.6 but i need to know if support for 3.7 will be provided in the near future?

#### Steps/Code to Reproduce
pip install pmdarima with python version 3.7

#### Versions
Linux-4.15.0-42-generic-x86_64-with-Ubuntu-18.04-bionic
Python 3.7.1 (default, Oct 22 2018, 11:21:55) 
[GCC 8.2.0]
NumPy 1.15.4
SciPy 1.1.0
Scikit-Learn 0.20.1
Statsmodels 0.9.0

### PR DESCRIPTION
**Pull request by astrojuanlu**
Fix #63.

### CODE REVIEW

**Comment by Juanlu001** [comment]
Didn't touch the Circle CI config yet, so I can only assume that the PyPy36 failure is unrelated.

**Comment by tgsmith61591** [comment]
I was seeing that issue on the latest sklearn on Travis before. I will try to figure out how I got around it in the past... Thanks for the initial PR!

File: appveyor.yml [hunk 1]
```diff
@@ -39,6 +39,12 @@ environment:
       PMDARIMA_MPL_DEBUG: "true"
       PYTHON: "C:\\Python36-x64"
 
+    - PYTHON_ARCH: "64"
+      PYTHON_VERSION: "3.7.1"
+      MPL_VERSION: "2.2.3"
+      PMDARIMA_MPL_DEBUG: "true"
+      PYTHON: "C:\\Python36-x64"
```
Comment (aaronreidsmith) [inline appveyor.yml]: I think this should be `"C:\\Python37-x64"`

Comment (Juanlu001) [reply appveyor.yml]: Good catch! Resolved

**Comment by Juanlu001** [comment]
Getting only one error in Python 3.7 on Linux:

> `numpy.linalg.linalg.LinAlgError: SVD did not converge in Linear Least Squares`

Not sure how to fix this one, and Google did not help. Ideas welcome :)

**Comment by tgsmith61591** [comment]
You were merging into `master`, so I'm guessing your branch was based on `master`. But we use gitflow, so `develop` is the preferred branch except for releases. I updated the destination branch and merged `develop` into yours so the PR is up-to-date, but anymore changes you make on remote may require you to pull from upstream.

**Pull request closed without merging**